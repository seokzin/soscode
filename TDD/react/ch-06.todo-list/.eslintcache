[{"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/index.tsx":"1","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/reportWebVitals.ts":"2","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/App.tsx":"3","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/index.tsx":"4","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Button/index.tsx":"5","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Input/index.tsx":"6","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/ToDoItem/index.tsx":"7"},{"size":501,"mtime":1608450377531,"results":"8","hashOfConfig":"9"},{"size":425,"mtime":1607145602318,"results":"10","hashOfConfig":"9"},{"size":1672,"mtime":1615884272454,"results":"11","hashOfConfig":"9"},{"size":79,"mtime":1608450376614,"results":"12","hashOfConfig":"9"},{"size":989,"mtime":1608450392527,"results":"13","hashOfConfig":"9"},{"size":644,"mtime":1610068351702,"results":"14","hashOfConfig":"9"},{"size":655,"mtime":1608450377481,"results":"15","hashOfConfig":"9"},{"filePath":"16","messages":"17","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},"1o6yy7c",{"filePath":"19","messages":"20","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"21","messages":"22","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"23","messages":"24","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"25","messages":"26","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"27","messages":"28","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"29"},{"filePath":"30","messages":"31","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/index.tsx",[],["32","33"],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/reportWebVitals.ts",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/App.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Button/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Input/index.tsx",[],["34","35"],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/ToDoItem/index.tsx",[],{"ruleId":"36","replacedBy":"37"},{"ruleId":"38","replacedBy":"39"},{"ruleId":"36","replacedBy":"40"},{"ruleId":"38","replacedBy":"41"},"no-native-reassign",["42"],"no-negated-in-lhs",["43"],["42"],["43"],"no-global-assign","no-unsafe-negation"]