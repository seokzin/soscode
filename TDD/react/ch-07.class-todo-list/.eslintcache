[{"/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/Button/index.tsx":"1","/Users/jeonghean/projects/react-book/code/class-todo-list/src/reportWebVitals.ts":"2","/Users/jeonghean/projects/react-book/code/class-todo-list/src/index.tsx":"3","/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/Input/index.tsx":"4","/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/ToDoItem/index.tsx":"5","/Users/jeonghean/projects/react-book/code/class-todo-list/src/App.tsx":"6"},{"size":1057,"mtime":1608810904690,"results":"7","hashOfConfig":"8"},{"size":425,"mtime":1607145602318,"results":"9","hashOfConfig":"8"},{"size":501,"mtime":1608450377531,"results":"10","hashOfConfig":"8"},{"size":706,"mtime":1608810377314,"results":"11","hashOfConfig":"8"},{"size":731,"mtime":1608810706026,"results":"12","hashOfConfig":"8"},{"size":2961,"mtime":1608882345348,"results":"13","hashOfConfig":"8"},{"filePath":"14","messages":"15","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1v95oxn",{"filePath":"16","messages":"17","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"18","messages":"19","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"20","messages":"21","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"22","messages":"23","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"24","messages":"25","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/Button/index.tsx",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/reportWebVitals.ts",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/index.tsx",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/Input/index.tsx",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/ToDoItem/index.tsx",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/App.tsx",[]]