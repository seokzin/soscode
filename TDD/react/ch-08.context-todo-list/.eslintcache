[{"/Users/jeonghean/projects/react-book/code/ch8.context-todo-list/src/index.tsx":"1","/Users/jeonghean/projects/react-book/code/ch8.context-todo-list/src/reportWebVitals.ts":"2","/Users/jeonghean/projects/react-book/code/ch8.context-todo-list/src/Contexts/ToDoList/index.tsx":"3"},{"size":501,"mtime":1609307758343,"results":"4","hashOfConfig":"5"},{"size":425,"mtime":1609307758380,"results":"6","hashOfConfig":"5"},{"size":1295,"mtime":1609756303346,"results":"7","hashOfConfig":"5"},{"filePath":"8","messages":"9","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"qr7ri0",{"filePath":"10","messages":"11","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"12","messages":"13","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/jeonghean/projects/react-book/code/ch8.context-todo-list/src/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch8.context-todo-list/src/reportWebVitals.ts",[],"/Users/jeonghean/projects/react-book/code/ch8.context-todo-list/src/Contexts/ToDoList/index.tsx",[]]