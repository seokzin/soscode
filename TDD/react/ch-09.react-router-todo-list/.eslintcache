[{"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/index.tsx":"1","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/ToDoList/index.tsx":"2","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/Detail/index.tsx":"3","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/Add/index.tsx":"4","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Contexts/ToDoList/index.tsx":"5","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/NotFound/index.tsx":"6","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/InputContainer/index.tsx":"7","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/PageHeader/index.tsx":"8"},{"size":590,"mtime":1611741896766,"results":"9","hashOfConfig":"10"},{"size":634,"mtime":1610777121363,"results":"11","hashOfConfig":"10"},{"size":1109,"mtime":1611366664483,"results":"12","hashOfConfig":"10"},{"size":565,"mtime":1610775384887,"results":"13","hashOfConfig":"10"},{"size":1295,"mtime":1610774495642,"results":"14","hashOfConfig":"10"},{"size":204,"mtime":1610952135141,"results":"15","hashOfConfig":"10"},{"size":833,"mtime":1610955003283,"results":"16","hashOfConfig":"10"},{"size":1023,"mtime":1611743640430,"results":"17","hashOfConfig":"10"},{"filePath":"18","messages":"19","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"20"},"7xskaz",{"filePath":"21","messages":"22","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"23","messages":"24","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"25","messages":"26","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"27","messages":"28","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"29","messages":"30","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"31","messages":"32","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"33","messages":"34","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/index.tsx",[],["35","36"],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/ToDoList/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/Detail/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/Add/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Contexts/ToDoList/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/NotFound/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/InputContainer/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/PageHeader/index.tsx",[],{"ruleId":"37","replacedBy":"38"},{"ruleId":"39","replacedBy":"40"},"no-native-reassign",["41"],"no-negated-in-lhs",["42"],"no-global-assign","no-unsafe-negation"]