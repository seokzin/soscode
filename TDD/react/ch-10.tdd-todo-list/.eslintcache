[{"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/App.tsx":"1","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/reportWebVitals.ts":"2","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/index.tsx":"3","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Components/PageHeader/index.tsx":"4","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Components/index.tsx":"5","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Components/Button/index.tsx":"6","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Pages/List/index.tsx":"7","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Pages/Add/index.tsx":"8","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Pages/Detail/index.tsx":"9","/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Pages/NotFound/index.tsx":"10"},{"size":396,"mtime":1613378867829,"results":"11","hashOfConfig":"12"},{"size":425,"mtime":1611895667453,"results":"13","hashOfConfig":"12"},{"size":590,"mtime":1612082462093,"results":"14","hashOfConfig":"12"},{"size":1024,"mtime":1612083209602,"results":"15","hashOfConfig":"12"},{"size":56,"mtime":1612425367373,"results":"16","hashOfConfig":"12"},{"size":988,"mtime":1612425737728,"results":"17","hashOfConfig":"12"},{"size":2169,"mtime":1612868300369,"results":"18","hashOfConfig":"12"},{"size":895,"mtime":1613119862638,"results":"19","hashOfConfig":"12"},{"size":1160,"mtime":1613286147862,"results":"20","hashOfConfig":"12"},{"size":204,"mtime":1613378805920,"results":"21","hashOfConfig":"12"},{"filePath":"22","messages":"23","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"h9uhk5",{"filePath":"24","messages":"25","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"26","messages":"27","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"28","messages":"29","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"30","messages":"31","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"32","messages":"33","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"34","messages":"35","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"36","messages":"37","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"38","messages":"39","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"40","messages":"41","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/App.tsx",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/reportWebVitals.ts",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/index.tsx",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Components/PageHeader/index.tsx",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Components/index.tsx",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Components/Button/index.tsx",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Pages/List/index.tsx",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Pages/Add/index.tsx",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Pages/Detail/index.tsx",[],"/Users/jeonghean/projects/react-book/code/tdd-todo-list/src/Pages/NotFound/index.tsx",[]]