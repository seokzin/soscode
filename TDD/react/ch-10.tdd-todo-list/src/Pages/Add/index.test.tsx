import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Router, useLocation } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import 'jest-styled-components';

import { Add } from './index';

describe('<Add />', () => {
  it('renders component correctly', () => {
    const history = createMemoryHistory();
    history.push('/add');

    const { container } = render(
      <Router history={history}>
        <Add />
      </Router>,
    );

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    expect(input).toBeInTheDocument();
    const button = screen.getByText('추가');
    expect(button).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('add a new ToDo and redirect to the root page', () => {
    localStorage.setItem('ToDoList', '["Old ToDo"]');

    const history = createMemoryHistory();
    history.push('/add');

    const TestComponent = (): JSX.Element => {
      const { pathname } = useLocation();
      return <div>{pathname}</div>;
    };

    render(
      <Router history={history}>
        <TestComponent />
        <Add />
      </Router>,
    );

    const url = screen.getByText('/add');
    expect(url).toBeInTheDocument();

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    const button = screen.getByText('추가');
    fireEvent.change(input, { target: { value: 'New ToDo' } });
    fireEvent.click(button);
    expect(localStorage.getItem('ToDoList')).toBe('["Old ToDo","New ToDo"]');
    expect(url.textContent).toBe('/');
  });

  it('do nothing if the input is empty', () => {
    localStorage.setItem('ToDoList', '["Old ToDo"]');

    const history = createMemoryHistory();
    history.push('/add');

    const TestComponent = (): JSX.Element => {
      const { pathname } = useLocation();
      return <div>{pathname}</div>;
    };

    render(
      <Router history={history}>
        <TestComponent />
        <Add />
      </Router>,
    );

    const url = screen.getByText('/add');
    expect(url).toBeInTheDocument();

    const button = screen.getByText('추가');
    fireEvent.click(button);

    expect(localStorage.getItem('ToDoList')).toBe('["Old ToDo"]');
    expect(url.textContent).toBe('/add');
  });
});
