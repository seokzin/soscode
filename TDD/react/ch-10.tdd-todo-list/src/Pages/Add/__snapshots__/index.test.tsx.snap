// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> renders component correctly 1`] = `
.c2 {
  text-align: center;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
  background-color: #304FFE;
}

.c2:hover {
  background-color: #1E40FF;
}

.c2:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c3 {
  color: #FFFFFF;
  font-size: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  font-size: 16px;
  padding: 10px 10px;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="할 일을 입력해 주세요"
    />
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        추가
      </div>
    </div>
  </div>
</div>
`;
